src/tests/state/experiments/selectors.test.js

```diff
@@ -1,13 +1,204 @@
-import { List, Map } from "immutable";
-import { getExperimentsCount } from "../../../state/experiments/selectors";
+import { fromJS } from "immutable";
+import { getExperimentsCount, getFilteredExperimentsByDate, 
+  getStartDatepickerTimestamp, getEndDatepickerTimestamp 
+} from "../../../state/experiments/selectors";
 
-describe("Selectors", () => {
+describe("getExperimentsCount test", () => {
   it("should get experiment count", () => {
-    let mockedState = Map({
-      experiments: Map({
-        items: List([1, 2, 3, 4])
-      })
+    let mockedState = fromJS({
+      experiments: {
+        items: [1, 2, 3, 4]
+      }
     });
     expect(getExperimentsCount(mockedState)).toEqual(4);
   });
 });
+
+describe("getFilteredExperimentsByDate tests", () => {
+  it("should return an immutable List of filtered experiments, where 1 experiment falls between startDate and endDate range", () => {
+    let mockedState = fromJS({
+      experiments: {
+        items: [
+          {
+            start_date: Date.parse("2018-08-22"),
+            end_date: Date.parse("2019-01-20"),
+            name: "test experiment name 1"
+          },
+          {
+            start_date: Date.parse("2017-08-22"),
+            end_date: Date.parse("2019-01-20"),
+            name: "test experiment name 2"
+          }
+        ]
+      }, 
+      dates: {
+        startDate: "2018-07-22",
+        endDate: "2019-07-22"
+      }
+    });
+    expect(getFilteredExperimentsByDate(mockedState)).toEqual(
+      fromJS([
+        {
+          "start_date": Date.parse("2018-08-22"),
```

Comment:
Sometimes it's helpful to setup some test data that will be shared across multiple tests and then the tests can be teensy tiny.  It looks like each test creates a few experiments, then filters, and looks for some of them and not others in the output?  Maybe you could make that shorter by moving them up a level and sharing them between the tests?

Reply:
that's a good idea 👍